src/formatDate.js

```diff
@@ -50,7 +53,65 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prefixYear = (yearBefore, newFormat) => {
+    const yearUpdateFormat
+      = newFormat.filter(partDate => partDate.includes('YY'))[0];
+
+    const previousCentury = '19';
+    const currentCentury = '20';
+
+    if (yearBefore.toString().length > yearUpdateFormat.length) {
+      return yearBefore.slice(-2);
+    }
+
+    if (yearBefore.length === yearUpdateFormat.length) {
+      return yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore >= 30) {
+      return previousCentury + yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore < 30) {
+      return currentCentury + yearBefore;
+    }
+
+    return yearBefore;
+  };
+
+  let newFormatDate = '';
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const dateArr = date.split(fromFormat[3]);
+    const indexYY = fromFormat.indexOf('YY') !== -1
+      ? fromFormat.indexOf('YY')
+      : fromFormat.indexOf('YYYY');
+
+    const indexMM = fromFormat.indexOf('MM');
+    const indexDD = fromFormat.indexOf('DD');
+
+    const day = dateArr[indexDD];
+    const month = dateArr[indexMM];
+    const year = dateArr[indexYY];
+
+    const separator = toFormat[toFormat.length - 1];
```

Comment:
> const dateArr = date.split(fromFormat[3]);

On 85th line you used last index, but here you are using length - 1, as for me it will be better to do it in one way(length - 1 in both lines or last index in both lines)

Reply:
agree